### ISSUE TITLE
heap buffer overflow in jpgicc.c

### ISSUE THREAD

**Issue by treebacker** [mm2/Little-CMS#258]
In source code `utils/jpgicc/jpgicc.c`, function `HandleEXIF`:

Below code:

```c
     // Search the directory for resolution tags
                numEntries = read16(data, ifd_ofs, swapBytes);

                for (i=0; i < numEntries; i++) {

                    uint32_t entryOffset = ifd_ofs + 2 + (12 * i);
                    uint32_t tag = read16(data, entryOffset, swapBytes);
     .....
```

`numEntries` may be a large number with a crafted file, then `entryOffset` is affected as a large number, results a heap overflow in `read16`.

 In AddressSanitizer:
![4](https://user-images.githubusercontent.com/44844446/115103538-2bd86200-9f85-11eb-9300-6b63baca7b5f.png)
In gdb:
![bug_4](https://user-images.githubusercontent.com/44844446/115103544-34309d00-9f85-11eb-8c95-551a43d3565e.png)

The crafted file is in the zip:
[bug4.zip](https://github.com/mm2/Little-CMS/files/6328689/bug4.zip)

**Comment by mm2** [comment]
Thanks for reporting.
However, what you are reporting may be a vulnerability of jpegicc, which is a **sample** of how to use the little CMS library and **it is not required or used** in client applications using the library. So, I will try to add some checks as long as that would not affect code readability, which is the primary goal of this sample.

 **Please note any CVE regarding this kind of vulnerabilities would be promptly disputed, as are related with samples and not with the little cms component, which is unaffected**.

**Comment by mm2** [comment]
Fixed per 8564c9b4fbf07d23062bbbbd87d5eb10dd92073f Thanks for reporting
